File: Tut_03/Sorting/Python/bubbleSort.py
def bubbleSort(arr):
    lenArr = len(arr)
    for i in range(lenArr):
        checked = False
        for j in range(0, lenArr - i - 1):
            if (arr[j] > arr[j + 1]):
                arr[j], arr[j+1] = arr[j+1], arr[j]
                checked = True
        if  not checked:
            break

arr = [3,21,5,1,6,4,7,3,1,4]
bubbleSort(arr)
print(arr)

File: Tut_03/Sorting/Javascript/bubbleSort.js
const bubbleSort = (a) => {
  let i = 0,
    j = 0,
    temp = 0;
  let len = arr.length;
  for (i = 0; i < len; i++) {
    let check = false;
    for (j = 0; j < len - 1 - i; j++) {
      if (a[j + 1] < a[j]) {
        temp = a[j];
        a[j] = a[j + 1];
        a[j + 1] = temp;
        check = true;
      }
    }
    if (!check) {
      break;
    }
  }
};

const arr = [4,2,41,5,14,1,1,12,5,12]
bubbleSort(arr)
console.log(arr)
